db/db_impl/db_impl.cc

```diff
@@ -5154,6 +5155,42 @@ Status DBImpl::TraceIteratorSeekForPrev(const uint32_t& cf_id, const Slice& key,
   return s;
 }
 
+Status DBImpl::DumpCache(const CacheDumpOptions& dump_options,
+                         const std::shared_ptr<Cache> cache) {
+  if (cache == nullptr) {
+    return Status::InvalidArgument("Cache pointer is null!");
+  }
+  CacheDumper dumper(env_, dump_options, cache, db_id_,
+                     immutable_db_options_.clock);
+  std::set<std::string> prefix_set;
```

Comment:
By the way, this could be a good application for temporary Bloom filter if you need to boost performance, even with 20 or 30 bits per key to minimize false inclusions.

Reply:
You mean, we can build the temporal bloom filter blocks independently and ingest to the block cache?

Reply:
I'm just suggesting that if doing std::set::find on each block cache key prefix is slow, as in dumping a DB using few entries in a shared block cache, using a Bloom filter created on those prefixes would be much faster. And while I was previously thinking we would just accept the Bloom filter result, we can obviously double-check the std::set whenever we have a Bloom hit, which is when we have to do more work anyway to serialize.

Sorry for diving into premature optimization.

Reply:
> I'm just suggesting that if doing std::set::find on each block cache key prefix is slow, as in dumping a DB using few entries in a shared block cache, using a Bloom filter created on those prefixes would be much faster. And while I was previously thinking we would just accept the Bloom filter result, we can obviously double-check the std::set whenever we have a Bloom hit, which is when we have to do more work anyway to serialize.
> 
> Sorry for diving into premature optimization.

This would be a really useful suggestion, especially when we use "sst_unqieu_id" as the prefix, the checking process may take time even we use std::set (theoritically it is O(1), but still slower than using BloomFilter). Also, I think it is tolerable that we dump out a few number of blocks that are the false positive from Bloom Filter.